tests/scenario/test_grafana_cloud_integrator_lib.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from charms.grafana_cloud_integrator.v0.cloud_config_requirer import (
+    CloudConfigAvailableEvent,
+    CloudConfigRevokedEvent,
+    GrafanaCloudConfigRequirer,
+)
+from ops import CharmBase, Framework
+from scenario import Context, Relation, State
+
+
+class MyCharm(CharmBase):
+    def __init__(self, framework: Framework):
+        super().__init__(framework)
+        self.cloud = GrafanaCloudConfigRequirer(self)
+
+
+@pytest.fixture()
+def context():
+    return Context(
+        charm_type=MyCharm,
+        meta={
+            "name": "my-charm",
+            "requires": {
+                "grafana-cloud-config": {"interface": "grafana_cloud_config", "limit": 1}
+            },
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "is_leader",
```

Comment:
it should be:
```
@pytest.mark.parametrize("is_leader", (True, False))
```

Reply:
(below too)

also something I often do is add a parametrized fixture containing a 'base state' which you can reuse in multiple tests, and put there the leadership parametrization.
So that all tests using that base state run with both leadership conditions.

If you need to make minor changes to that state you can use the `.replace` API, if the changes are major, create a state from scratch.

Reply:
example: https://github.com/canonical/cos-lib/pull/55/files#diff-fec86a78873f1e97b39f65ce6ce9869b5c4a351e99aa4ba09495f42a8d061a5dR54

Reply:
this may be the first time I've ever parameterized a fixture in so few characters that it could be on one line

Since its only used twice and in anticipation of @sed-i citing clarity as an objection to putting the parametrization into the fixtures, I've not added parameters into the fixtures for now.  But if you have a strong request for them, I'm happy to add.

Reply:
Resolving given @Abuelodelanada's review, but if anyone disagrees we can open another issue and fix it